### PR TITLE
Adds a condition for is divisible by

### PR DESCRIPTION
**Pull request by EquipableMC**
### Description
This PR simply adds a condition to check if a number is divisible by another number, without having to use the mod function.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
```
Comment (Asleeepp) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:10]: docs?

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%number% is divisible by %number%",
+			"%number% can be divisible by %number%");
```
Comment (Asleeepp) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:16]:
```suggestion
			"%number% can be divided by %number%");
```

**Review by Asleeepp — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 2 above)
Comment (Efnilite) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:16]: this should be one syntax

**Review by Efnilite — CHANGES_REQUESTED** [review]
and tests!

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 2 above)
Comment (Asleeepp) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:16]: i was thinking of saying that too

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		if (divisor == 0) {
+			return false; // Avoid division by zero
+		}
+
+		return number1.doubleValue() % divisor == 0;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "is divisible by " + num1.toString(event, debug) + num2.toString(event, debug);
```
Comment (Romitou) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:51]:
```suggestion
		return num1.toString(event, debug) + " is divisible by " + num2.toString(event, debug);
```

**Comment by EquipableMC** [comment]
> and tests!

😭 okay

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 import org.bukkit.event.Event;
 import org.jetbrains.annotations.Nullable;
 
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
+@Examples("5 is divisible by 5")
```
Comment (Asleeepp) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:17]:
```suggestion
@Examples("if 5 is divisible by 5:")
```
id rather there be more examples, show off the other syntax too

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+test "is divisible":
+	assert 5 can be divided by 5 with "5 can be divided by 5!"
+	assert 5 can be divided by 10 to fail with "5 cannot be divided by 10!"
```
Comment (Asleeepp) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:3]:
```suggestion
assert 5 can be divided by 10 to fail with "5 cannot be divided by 10!"
```
needs more imo, test like really high numbers

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
+@Examples("5 is divisible by 5")
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:17]: should also include an example that returns false for this
`11 is divisible by 10 # false`

**Review by Asleeepp — CHANGES_REQUESTED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk (hunk 5 above)
Comment (Asleeepp) [reply src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:3]: whoops, i screwed up that suggestion

Comment (Asleeepp) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:3]:
```suggestion
	assert 5 can be divided by 10 to fail with "5 cannot be divided by 10!"
	
```

**Review by Asleeepp — CHANGES_REQUESTED** [review]
fix previous suggestion

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+		Number number2 = num2.getSingle(event);
+
+		if (number1 == null || number2 == null) {
+			return false; // Prevent NullPointerException
+		}
+
+		double divisor = number2.doubleValue();
+
+		if (divisor == 0) {
+			return false; // Avoid division by zero
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:50]: i'm not sure this should return false as this is a mathematical error and so perhaps should also throw or at least log something. but idk

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk (hunk 5 above)
Comment (Efnilite) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:3]:
```suggestion
	assert 5 can be divided by 10 to fail with "5 cannot be divided by 10!"

```

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk
Comment (Efnilite) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:1]: you should include that 
- expr 1 and/or expr 2 are none that it fails
- if you put in 0 it also fails
- expr 1 being bigger than expr 2, not just equal

:)

**Review by Efnilite — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 7 above)
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:50]: then use mod function at this point, this is just a "hey this is true so it's correct" or "hey this is false so it isn't correct"

Comment (Efnilite) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:50]: yeah ig

Comment (TenFont) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:50]: I don't think an error should be thrown here, **for now**. Until we have a system for runtime errors, this should just return false. Similarly issue in #7038

**Comment by EquipableMC** [comment]
> I believe a negated version of the pattern should exist. Something like `if {_number} is not divisible by 8:` should be possible.

added!

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 8]
```diff
@@ -14,13 +14,17 @@
...
+		"if 11 cannot be divided by 10:",
+		"if 10007 cannnot be divided by 10007:"})
 @Since("INSERT VERSION")
 public class CondIsDivisibleBy extends Condition {
 
 	static {
 		Skript.registerCondition(CondIsDivisibleBy.class,
-			"%number% (is|can be) (divisible|divided) by %number%");
+			"%number% (is|can be) (divisible|divided) by %number%",
+			"%number% (isn't|is not|can[ ]not be) (divisible|divided) by %number%");
```
Comment (Asleeepp) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:27]: wheres the negated logic? you cant just add negated syntax and expect it to automatically work

**Review by Asleeepp — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 8 above)
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:27]: https://github.com/SkriptLang/Skript/pull/7104#discussion_r1769552152

Comment (Asleeepp) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:27]: ?? that doesnt relate to anything i just said

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 9]
```diff
@@ -0,0 +1,7 @@
+test "is divisible":
+	assert 5 can be divided by 5 with "5 can be divided by 5!"
+	assert 5 can not be divided by 10 to fail with "5 cannot be divided by 10!"
```
Comment (sovdeeth) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:3]: Why is this asserting to fail
It should be true

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+		"if 1964903306 is divisible by 982451653:",
+		"if 11 cannot be divided by 10:",
+		"if 10007 cannnot be divided by 10007:"})
+@Since("INSERT VERSION")
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%number% (is|can be) (divisible|divided) by %number%",
+			"%number% (isn't|is not|can[ ]not be) (divisible|divided) by %number%");
```
Comment (TenFont) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:27]: These patterns should be changed, currently they allow for things like `is divided by` and `can't be divisible by` which don't make sense.

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 11]
```diff
@@ -0,0 +1,7 @@
+test "is divisible":
+	assert 5 can be divided by 5 with "5 can be divided by 5!"
+	assert 5 can not be divided by 10 with "5 cannot be divided by 10!"
+	assert 5 isn't divisible by 0 with "nothing can be divided by 0!"
+	assert 1964903306 is divisible by 982451653 with "1964903306 is divisible by 982451653!"
+	set {_none} to ("" parsed as int)
```
Comment (Pikachu920) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:6]: the var will be none by default, so no need to set it

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	@Override
+	public boolean check(final Event event) {
+		Number number2 = num2.getSingle(event);
+		return num1.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number number1) {
+				double divided = number1.doubleValue();
+				double divisor = number2 != null ? number2.doubleValue() : 0;
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:53]: if the provided value is null, we should return false rather than defaulting to 0

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		num1 = (Expression<Number>) exprs[0];
+		num2 = (Expression<Number>) exprs[1];
+		setNegated(matchedPattern == 1 || matchedPattern == 3);
+		return true;
+	}
+
+	@Override
+	public boolean check(final Event event) {
+		Number number2 = num2.getSingle(event);
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:48]: why not use divisor/dividend insteas of number1/2

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+		"if 11 cannot be divided by 10:",
+		"if 10007 cannnot be divided by 10007:"})
+@Since("INSERT VERSION")
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%numbers% (is|are) divisible by %number%",
+			"%numbers% (isn't|is not|aren't|are not) divisible by %number%",
+			"%numbers% can be [evenly] divided by %number%",
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:29]: should evenly be optional?

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+			"%numbers% (can't|cannot|can not) be [evenly] divided by %number%");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<Number> num1;
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:34]: why not divisorExpression for example

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 16]
```diff
@@ -0,0 +1,7 @@
+test "is divisible":
```
Comment (Pikachu920) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:1]: woooo tests

**Review by Pikachu920 — CHANGES_REQUESTED** [review]
yay english

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 12 above)
Comment (Pikachu920) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:53]: this can also be extracted to  outside the checker

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 14 above)
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:29]: Eh it should be left optional in my opinion.

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividedExpression = (Expression<Number>) exprs[1];
+		setNegated(matchedPattern == 1 || matchedPattern == 3);
+		return true;
+	}
+
+	@Override
+	public boolean check(final Event event) {
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:47]:
```suggestion
	public boolean check(Event event) {
```

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
+@Examples({"if 5 is divisible by 5:",
+		"if 1964903306 is divisible by 982451653:",
+		"if 11 cannot be divided by 10:",
+		"if 10007 cannnot be divided by 10007:"})
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:21]:
```suggestion
@Examples({
	"if 5 is divisible by 5:",
	"if 11 cannot be divided by 10:",
})
```

**Review by Efnilite — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 14 above)
Comment (Pikachu920) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:29]: how come? it is a bit confusing to me that `if 2 is divisible by 3` is false. it is divisible, but its not evenly divisible

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividedExpression = (Expression<Number>) exprs[1];
+		setNegated(matchedPattern == 1 || matchedPattern == 3);
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number number2 = dividedExpression.getSingle(event);
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:48]: still needs a rename :D

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+		return divisorExpression.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number number) {
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:51]: also needs a rename :D

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 14 above)
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:29]: Valid point

**Review by Pikachu920 — CHANGES_REQUESTED** [review]

**Review by sovdeeth — CHANGES_REQUESTED** [review]
tests failing

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+})
+@Since("INSERT VERSION")
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%numbers% (is|are) divisible by %number%",
+			"%numbers% (isn't|is not|aren't|are not) divisible by %number%",
+			"%numbers% can be evenly divided by %number%",
+			"%numbers% (can't|cannot|can not) be [evenly] divided by %number%");
```
Comment (Phill310) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:30]: I feel like the use of `evenly` should be consistent throughout all of the syntax. I think it should be required for all syntaxes but if you want to make it optional that would be okay too.

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+	@SuppressWarnings("null")
+	private Expression<Number> divisorExpression;
+	@SuppressWarnings("null")
+	private Expression<Number> dividedExpression;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividedExpression = (Expression<Number>) exprs[1];
```
Comment (Phill310) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:41]: You forgot the n in dividend

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 23]
```diff
@@ -0,0 +1,6 @@
...
+	assert 5 can be divided by 5 with "5 can be divided by 5!"
```
Comment (Phill310) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:2]: I am guessing you forgot to update this when you added the `evenly` to the syntax

**Review by Phill310 — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java (hunk 21 above)
Comment (Pikachu920) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:30]: required in all please :D

**Review by Pikachu920 — CHANGES_REQUESTED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk (hunk 23 above)
Comment (EquipableMC) [reply src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:2]: yeah oops

**Review by Efnilite — APPROVED** [review]

**Review by Phill310 — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number divisorNumber = dividendExpression.getSingle(event);
+		return divisorExpression.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number dividedNumber) {
+				double divided = dividedNumber.doubleValue();
```
[the remark this answers is not in the document]
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:52]: yeah but I have to make sure it's actually `dividend` and not `divided`

[the remark this answers is not in the document]
Comment (Phill310) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:52]: It shouldn't ever be `divided`. The proper term for the number being divided is `dividend`.

[the remark this answers is not in the document]
Comment (EquipableMC) [reply src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:52]: oh alright gotcha.

**Comment by EquipableMC** [comment]
I do not understand why tests are failing, if someone can help point it out that would be nice

**Comment by TenFont** [comment]
> I do not understand why tests are failing, if someone can help point it out that would be nice

You should read the build logs
```
Failed:
  is divisible (on 2 environments)
    5 cannot be divided by 10! [CondIsDivisibleBy.sk, line 3] (on paper-1.20.6)
    5 cannot be divided by 10! [CondIsDivisibleBy.sk, line 3] (on paper-1.21.0)
```

**Comment by EquipableMC** [comment]
> > I do not understand why tests are failing, if someone can help point it out that would be nice
> 
> 
> 
> You should read the build logs
> 
> ```
> 
> Failed:
> 
>   is divisible (on 2 environments)
> 
>     5 cannot be divided by 10! [CondIsDivisibleBy.sk, line 3] (on paper-1.20.6)
> 
>     5 cannot be divided by 10! [CondIsDivisibleBy.sk, line 3] (on paper-1.21.0)
> 
> ```

I have, it just doesn't tell me WHY it's failing.

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividendExpression = (Expression<Number>) exprs[1];
+		setNegated(matchedPattern == 1 || matchedPattern == 3);
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number divisorNumber = divisorExpression.getSingle(event);
+		return divisorExpression.check(event, new Checker<Number>() {
```
Comment (TenFont) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:49]: This check should be on the **dividendExpression**
You can also just use a lambda here

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+
+	@SuppressWarnings("null")
+	private Expression<Number> divisorExpression;
+	@SuppressWarnings("null")
+	private Expression<Number> dividendExpression;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividendExpression = (Expression<Number>) exprs[1];
```
Comment (TenFont) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:41]:
```suggestion
		dividendExpression = (Expression<Number>) exprs[0];
		divisorExpression = (Expression<Number>) exprs[1];
```
These need to be flipped around.

**Review by TenFont — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number divisorNumber = divisorExpression.getSingle(event);
+		return divisorExpression.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number dividendNumber) {
+				double dividend = dividendNumber.doubleValue();
+				double divisor = divisorNumber != null ? divisorNumber.doubleValue() : null;
```
Comment (TenFont) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:53]: If divisorNumber is null, you're still trying to unbox it into a double, which will throw a NullPointerException.
Essentially, regardless of whether or not you do the null check, the `doubleValue()` method is still being called!

**Review by TenFont — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Checker;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
```
Comment (Phill310) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:17]: Can we add `evenly` to the description? Maybe the name too

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+				double dividend = dividendNumber.doubleValue();
+				double divisor = divisorNumber.doubleValue();
+				return dividend % divisor == 0;
+			}
+		}, isNegated());
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return divisorExpression.toString(event, debug) + " is divisible by " + dividendExpression.toString(event, debug);
```
Comment (Phill310) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:60]: The toString should reflect whether the condition is negated. Also add `evenly`

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 30]
```diff
@@ -0,0 +1,6 @@
+test "is divisible":
+	assert 5 can be evenly divided by 5 with "5 can be divided by 5!"
+	assert 5 can not be evenly divided by 10 with "5 cannot be divided by 10!"
+	assert 5 isn't evenly divisible by 0 with "nothing can be divided by 0!"
+	assert 1964903306 is evenly divisible by 982451653 with "1964903306 is divisible by 982451653!"
+	assert {_none} is evenly divisible by 10 to fail with "you cannot divide by <none>!"
```
Comment (Phill310) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:6]: Let's add some tests where the first input is a list (since the syntax supports that)

**Review by Phill310 — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 31]
```diff
@@ -0,0 +1,64 @@
...
+				double divisor = divisorNumber.doubleValue();
+				return dividend % divisor == 0;
+			}
+		}, isNegated());
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return dividend.toString(event, debug) + " is " + (isNegated() ? "not " : "")
+			+ "evenly divisible by" + divisor.toString(event, debug);
```
Comment (Phill310) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:61]: Forgot a space after by :)

**Review by Phill310 — APPROVED** [review]
Looks alright overall

**Review by Romitou — APPROVED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 32]
```diff
@@ -0,0 +1,8 @@
...
+	assert 5, 10, 15 can be evenly divided by 5 with "5, 10, and 15 can be evenly divided by 5!"
+	assert 3, 5, 7 cannot be evenly divided by 2 with "3, 5, and 7 cannot be evenly divided by 2!"
```
Comment (sovdeeth) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:8]:
```suggestion
	assert 5, 10, and 15 can be evenly divided by 5 with "5, 10, and 15 can be evenly divided by 5!"
	assert 3, 5, and 7 cannot be evenly divided by 2 with "3, 5, and 7 cannot be evenly divided by 2!"
```
an OR list would also be good.

File: src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java [hunk 33]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number divisorNumber = divisor.getSingle(event);
+		return dividend.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number dividendNumber) {
+				double dividend = dividendNumber.doubleValue();
+				double divisor = divisorNumber.doubleValue();
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java:52]: why are we getting the divisor double value every time?

**Review by sovdeeth — CHANGES_REQUESTED** [review]

**Review by sovdeeth — APPROVED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk [hunk 34]
```diff
@@ -0,0 +1,10 @@
+test "is divisible":
```
Comment (Asleeepp) [inline src/test/skript/tests/syntaxes/conditions/CondIsDivisibleBy.sk:1]: i think this should have more edge cases (ex. strings or negative numbers)

**Review by Asleeepp — CHANGES_REQUESTED** [review]

**Review by TenFont — APPROVED** [review]

**Review by Asleeepp — APPROVED** [review]

**Review by Pikachu920 — APPROVED** [review]
